server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1016,10 +1024,18 @@
             sc.setParameters("serviceOfferingId", serviceOffId);
         }
 
+        if(securityGroupId != null) {
+            sc.setParameters("securityGroupId", securityGroupId);
+        }
+
         if (display != null) {
             sc.setParameters("display", display);
         }
 
+        if (display != null) {
```

Comment:
@GabrielBrascher I guess this is supposed to be isHaEnabled != null ?

Reply:
good catch @Pearl1594

Reply:
The code has been updated addressing both comments. Thanks for the review @Pearl1594 and @sureshanaparti.